site/pages/pledge/[address].tsx

```diff
@@ -83,6 +96,62 @@ export const getStaticProps: GetStaticProps<PageProps, Params> = async (
       pledge = { ...DEFAULT_VALUES, ownerAddress: resolvedAddress };
     }
 
+    // add up retirements here
+    let retirements: RetirementsTotalsAndBalances;
+    if (pledge.wallets && Object.values(pledge.wallets).length) {
+      const verifiedWallets = Object.values(pledge.wallets).filter(
+        (wallet) => wallet.status === "verified"
+      );
+      const promises = verifiedWallets.map((wallet) =>
+        getRetirementTotalsAndBalances({
+          address: wallet.address,
+        })
+      );
+      promises.push(
+        getRetirementTotalsAndBalances({
+          address: resolvedAddress,
+        })
+      );
+      const values: RetirementsTotalsAndBalances[] = await Promise.all(
+        promises
+      );
+      values.reduce((prev: any, curr) => {
+        if (prev) {
+          prev.totalRetirements = (
+            Number(prev.totalRetirements) + Number(curr.totalRetirements)
+          ).toString();
+          prev.totalTonnesRetired = (
+            Number(prev.totalTonnesRetired) + Number(curr.totalTonnesRetired)
+          ).toString();
+          prev.totalTonnesClaimedForNFTS = (
+            Number(prev.totalTonnesClaimedForNFTS) +
+            Number(curr.totalTonnesClaimedForNFTS)
+          ).toString();
+          prev.bct = (Number(prev.bct) + Number(curr.bct)).toString();
+          prev.mco2 = (Number(prev.mco2) + Number(curr.mco2)).toString();
+          prev.nct = (Number(prev.nct) + Number(curr.nct)).toString();
+          prev.ubo = (Number(prev.ubo) + Number(curr.ubo)).toString();
+          prev.nbo = (Number(prev.nbo) + Number(curr.nbo)).toString();
+
+          return prev;
+        } else {
+          return curr;
+        }
+      });
```

Comment:
should give it an initial value of empty object `{}`, this will help typescript understand our intention too.

```ts
const const sumValues = values.reduce<SumValuesType>((prev, curr) => {
   ...dostuff();
}, {} as SumValuesType)
```

note that this changes the behavior of reduce: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce
```
If initialValue is specified, that also causes currentValue to be initialized to the first value in the array. If initialValue is not specified, accumulator is initialized to the first value in the array, and currentValue is initialized to the second value in the array.
```